_targets.R

```diff
@@ -76,16 +79,85 @@ min_obs_days <- 100
 # Change dummy date to force re-build of NWIS DO sites and data download
 dummy_date <- "2022-06-15"
 
-# test and validation sites
+
+#2) Configure model inputs/variables 
+
+# Define test and validation sites
 val_sites <- c("01472104", "01473500", "01481500")
 tst_sites <- c("01475530", "01475548")
 
 train_start_date <- '1980-01-01'
-train_end_date <- '2017-01-01'
-val_start_date <- '2017-01-01'
-val_end_date <- '2019-01-01'
-test_start_date <- '2019-01-01'
-test_end_date <- '2022-01-01'
+train_end_date <- '2014-10-01'
+val_start_date <- '2014-10-01'
+val_end_date <- '2015-10-01'
+test_start_date <- '2015-10-01'
+test_end_date <- '2022-10-01'
+
+# Define model parameters and combine within a list that gets used to
+# write a base model config file for the snakemake modeling workflow.
+base_config_options <- list(
+  out_dir = "../../../out/models",
+  # random seed for training; If FALSE, no seed. Otherwise, specify the seed:
+  seed = FALSE,
+  num_replicates = 1,
+  trn_offset = 1,
+  tst_val_offset = 1,
+  epochs = 100,
+  hidden_size = 10,
+  dropout = 0.2,
+  recurrent_dropout = 0.2,
+  finetune_learning_rate = 0.01,
+  early_stopping = FALSE,
+  # train/val/test split information is defined above:
+  val_sites = val_sites, 
+  test_sites = tst_sites,
+  train_start_date = train_start_date, 
+  train_end_date = train_end_date, 
+  val_start_date = val_start_date, 
+  val_end_date = val_end_date,
+  test_start_date = test_start_date, 
+  test_end_date = test_end_date
+  )
+
+# Define global model parameters for the "baseline" deep learning model
+x_vars_global <- c("pr","SLOPE","tmmx","tmmn","srad","CAT_BASIN_SLOPE","CAT_ELEV_MEAN",
```

Comment:
There may be a case where we decide to change `x_vars` for a particular model. I suggest still putting in the `base_config_options`. If there is a case where we want to change it for one model, we can provide a `x_vars` option in that model's configuration list and it will override the base_config's config option.

Reply:
Oh yeah, good point. I've added `x_vars_global` to the `base_config_options` list in https://github.com/USGS-R/drb-do-ml/pull/163/commits/8274cd1371abfaba6e767865cd7291b2aade0f5d.